explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-* Tricky: The API is tied to notifications. You can't show a badge unless there
-  are pending notifications.
+* A "flag" indicating the presence of a badge with no contents, or
+* A positive integer.
 
-### Chrome OS
+The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).
 
-Not currently supported, but will [soon be available](https://crbug.com/801014)
-for Android apps using the above `NotificationChannel` API. The Chrome browser
-itself will be able to re-use this mechanism for showing a coloured dot on web
-application icons.
+### The API
 
-### iOS
+The `Badge` interface is a member object on
+[`Window`](https://html.spec.whatwg.org/#the-window-object) and
+[`ServiceWorkerGlobalScope`](https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope-interface).
+It contains the following methods:
 
-* The host API is
-  [`UIApplication.applicationIconBadgeNumber`](https://developer.apple.com/documentation/uikit/uiapplication/1622918-applicationiconbadgenumber),
-  which lets you set a positive integer only.
-* Integer badges are passed directly to the host API.
-* Flag badges and strings other than digits are just represented by the number
-  "1".
+* `Badge.set([contents], [options])`: Sets the badge for the scope in *options*
+  to *contents* (an integer), or to "flag" if *contents* is omitted. If
+  *contents* is 0, clears the badge for the given scope.
+* `Badge.clear([options])`: Clears the badge for the scope in *options*.
+* `Badge.canBadgeDocument()`: Returns true if the user agent would display a
+  badge applying to a document scope on or near the page's favicon.
 
-### Ubuntu
+The *options* parameter is a dictionary containing a single member, `scope`,
+which contains a URL prefix to scope the badge to. If omitted, it defaults to
+`"/"`.
 
-Requires Ubuntu (no general API for Linux).
+> Note: Should we have a separate overload for boolean flags now, as discussed in [Issue 19](https://github.com/WICG/badging/issues/19) and [Issue 42](https://github.com/WICG/badging/issues/42)?
 
-* The host API is
-  [`unity_launcher_*`](https://wiki.ubuntu.com/Unity/LauncherAPI), which lets
-  you set an integer only.
-* See iOS treatment.
+### UX treatment
+Badges may appear in any place that the user agent deems appropriate. In general, these places should be obviously related to the pages being badged, so users understand what the status is for. Appropriate places could include:
+- Tab favicons.
+- Bookmark icons.
+- [OS Specific Contexts](#OS-Specific-Contexts) for [Installed Web Applications](https://www.w3.org/TR/appmanifest/#installable-web-applications).
 
-### Summary
+> Note: When showing an badge in an [OS Specific Context](#OS-Specific-Contexts) user agents should attempt reuse existing [operating system APIs and conventions](#Specific-operating-system-treatment-for-installed-web-applications), to achieve a native look-and-feel.
```

Comment:
Link looks broken to me. Should maybe be to `docs/implementation.md`

Reply:
Done.